experiments/batchOscillation/wf.yml

```diff
@@ -0,0 +1,10 @@
+steps:
+  - id: install-python-modules
+    uses: "ivotron/python-actions@master"
+    args:
+      - "pip install -r /workspace/scripts/requirements.txt"
+
+  - id: run-simulator
+    uses: "ivotron/python-actions@master"
+    args:
+      - "python /workspace/scripts/radossim.py"
```

Comment:
Using the official python docker image and explicitly dealing with the virtualenv could help:

```yaml
steps:
- id: install-requirements
  uses: docker://python:3.8-slim-buster
  runs: [bash, -uec]
  args:
  - |
    python -mvenv venv/
    source venv/bin/activate
    pip install -r scripts/requirements.txt

- id: run-sim
  uses: docker://python:3.8-slim-buster
  runs: [bash, -uec]
  args:
  - |
    source ./venv/bin/activate
    python scripts/radossim.py
```

the first step builds a virtual environment on the root of the project (on `.venv/` folder) and installs requirements there. The second loads the environment before running the script.

Reply:
and the above runs from the root of the project:

```bash
popper run -f experiments/batchOscillation/wf.yml
```

Reply:
Ah, I didn't know about `- |`. Also, how did you select `3.8-slim-buster` of all the tags available for python on docker hub? Or more generally, how should I determine what tag to use for any image?

Reply:
in general there are three types of official images, those based on 1) full debian, 2) debian-slim and 3) alpine. I usually use the latest stable release of python (3.8 as of today), and start with alpine cause it has a tiny footprint compared to the debian images. If requirements make use of system libraries though (e.g. numpy), alpine usually doesn't work because there are no pre-compiled binaries (wheels) for it, so then I move to the slim variant of debian-based images

Reply:
Ah, that’s super helpful - not only for python images. Can you add that to the popper documentation? Practical strategies on how to select images: use the latest version (survives the longest), use the smallest Linux image available that has the right wheels attached, and how do you find out it’s a missing wheel and not some other error that prevents things from working.